# HG changeset patch
# User Rik <rik@octave.org>
# Date 1459547812 25200
#      Fri Apr 01 14:56:52 2016 -0700
# Node ID 0d1bd2ede0c151f80d6d5b33323542cbcec93fe5
# Parent  683a1beee538fda9347183877d54c60442f0897b
doc: Document that logical negation is written with a space ("! A").

* contrib.txi: Document that logical negation is written with a space "! A".

diff --git a/doc/interpreter/contrib.txi b/doc/interpreter/contrib.txi
--- a/doc/interpreter/contrib.txi
+++ b/doc/interpreter/contrib.txi
@@ -356,18 +356,22 @@ Do not do this, however, with the iterat
 @group
 for i = 1:n
   b(i) = sum (a(:,i));
 endfor
 @end group
 @end example
 
 The Octave operator @samp{!} should be used for logical negation, rather than
-@samp{~}.  Comments should begin with the @samp{#} character, rather than
-@samp{%}.
+@samp{~}.  The negation operator is written with a space between the operator
+and its target, e.g., @code{! A}.  Comments should begin with the @samp{#}
+character, rather than @samp{%}.
+
+Any Built-In Self Tests or demos using the @code{%!test} or @code{%!demo}
+syntax should begin two lines after the @code{endfunction} keyword.
 
 @node C++ Sources
 @section C++ Sources
 
 Don't use tabs.  Tabs cause trouble.  If you are used to them, set up
 your editor so that it converts tabs to spaces.
 
 Format function headers like this:
@@ -423,16 +427,19 @@ SVD::type type = ((nargout == 0 || nargo
 @end group
 @end example
 
 @noindent
 Consider putting extra braces around a multi-line expression to make it
 more readable, even if they are not necessary.  Also, do not hesitate to
 put extra braces anywhere if it improves clarity.
 
+The negation operator is written with a space between the operator
+and its target, e.g., @code{! A}.
+
 Declare variables just before they are needed.  Use variables local to
 blocks---it helps optimization.  Don't write a multi-line variable
 declaration with a single type specification and multiple variables.  If
 the variables don't fit on single line, repeat the type specification.
 Example:
 
 @example
 @group
